### PR TITLE
SpringBoot - Add registered workflow and activity impl info to workers template

### PR DESCRIPTION
**Pull request by tsurdilo**
Update to WorkersTemplate to expose registered workflow and activity class info per task queue. 
Allows users to build custom actuator endpoints that display this.

Sample that uses this info to create a custom actuator endpoint https://github.com/temporalio/samples-java/pull/576

### CODE REVIEW

**Comment by tsurdilo** [comment]
Sample screenshot:

Screenshot 2024-02-11 at 7 30 23 PM

File: temporal-spring-boot-autoconfigure-alpha/src/main/java/io/temporal/spring/boot/autoconfigure/template/WorkersTemplate.java [hunk 1]
```diff
@@ -410,6 +426,47 @@ private Worker createNewWorker(
...
+      registeredActivityInfo.add(activityClass);
+      return this;
+    }
+
+    public RegisteredInfo addWorkflowInfo(String workflowClass) {
+      registeredWorkflowInfo.add(workflowClass);
+      return this;
+    }
+
+    public List<String> getRegisteredActivityInfo() {
```
Comment (Quinn-With-Two-Ns) [inline temporal-spring-boot-autoconfigure-alpha/src/main/java/io/temporal/spring/boot/autoconfigure/template/WorkersTemplate.java:461]: Do you think there is other information users may want us to expose in the future about activities or workflows other then their name?

Comment (Quinn-With-Two-Ns) [reply temporal-spring-boot-autoconfigure-alpha/src/main/java/io/temporal/spring/boot/autoconfigure/template/WorkersTemplate.java:461]: May make sense to return a list of objects so we can add new fields if decided to expose more info.

Comment (tsurdilo) [reply temporal-spring-boot-autoconfigure-alpha/src/main/java/io/temporal/spring/boot/autoconfigure/template/WorkersTemplate.java:461]: ok good point, will update

File: temporal-spring-boot-autoconfigure-alpha/src/main/java/io/temporal/spring/boot/autoconfigure/template/WorkersTemplate.java [hunk 2]
```diff
@@ -111,6 +112,13 @@ public Collection<Worker> getWorkers() {
     return workers;
   }
 
+  public Map<String, RegisteredInfo> getRegisteredInfo() {
```
Comment (Quinn-With-Two-Ns) [inline temporal-spring-boot-autoconfigure-alpha/src/main/java/io/temporal/spring/boot/autoconfigure/template/WorkersTemplate.java:115]: Does this need javadocs?

Comment (tsurdilo) [reply temporal-spring-boot-autoconfigure-alpha/src/main/java/io/temporal/spring/boot/autoconfigure/template/WorkersTemplate.java:116]: done

**Comment by tsurdilo** [comment]
@Quinn-With-Two-Ns made updates, let me know if ok now

**Review by Quinn-With-Two-Ns — CHANGES_REQUESTED** [review]
Can we mark this api as `@Experimental`

**Comment by tsurdilo** [comment]
Added Experimental flag to getRegisteredInfo method

**Pull request merged**